lib/message.rb

```diff
@@ -0,0 +1,38 @@
+require './lib/character'
+
+class Message
+  attr_reader :message_array, :shift
+
+  def initialize(message, shift)
+    @characters = ("a".."z").to_a << " "
+    @message_array = message.chars
+    @shift = shift
+  end
+
+  def move_alphabet(character)
+    @characters.rotate(@characters.index(character.downcase))
+  end
+
+  def change_message
+    @message_array.map! do |character|
+      Character.new(character)
+    end
+    @message_array.map do |character|
+      if @characters.index(character.id.downcase) == nil
+        character.id
+      else
+        find_new_char = move_alphabet(character.id)
+        if (@message_array.index(character) + 4) % 4 == 0
+          find_new_char.rotate(@shift[:A])[0]
+        elsif (@message_array.index(character) + 3) % 4 == 0
+          find_new_char.rotate(@shift[:B])[0]
+        elsif (@message_array.index(character) + 2) % 4 == 0
+          find_new_char.rotate(@shift[:C])[0]
+        else
+          find_new_char.rotate(@shift[:D])[0]
+        end
+      end
+    end.join
+  end
+
+end
```

Comment:
Turns every character in a message into an array of characters and then turns each character into an object that can be indexed. This then iterates through each index to shift it the appropriate amount. Then combines them together to make a new string.